api/internal/localizer/localizer_test.go

```diff
@@ -1540,6 +1540,43 @@ valuesFile: mapleValues.yaml
 	checkLocalizeInTargetSuccess(t, files)
 }
 
+func TestLocalizeGeneratorsHelmV1Beta1(t *testing.T) {
```

Comment:
The issue https://github.com/kubernetes-sigs/kustomize/issues/5083 reported an issue with `kustomize build`, why do we need to add a test for `localize` here? 

On that note, we should have a test for `kustomize build` somewhere in one of the files under api/krusty. You can take a look at some of the other tests in that file (e.g. [this one](https://github.com/kubernetes-sigs/kustomize/blob/56d37acc7d9a1484ea3d9cb54e12c47b2db635fc/api/krusty/baseandoverlaysmall_test.go#L260)) to see how to structure our `kustomize build` integration tests. I think we should use the exact content from https://github.com/kubernetes-sigs/kustomize/issues/5083 (adjusted to include what you needed to fix to reproduce the problem) as a test.